### ISSUE TITLE
How to add custom backward gradients

### ISSUE THREAD

**Issue by cdcseacave** [nerfstudio-project/gsplat#465]
Great project, thank you! This is more a cry for help. I'm trying to add new functionality to the forward renderer, which works great, but next I have huge problems understanding how to write the backward pass. 

Do you have any suggestions on how to best approach this task?

**Comment by YuxinYao620** [comment]
Hi cdcseacave, may I ask have you resolved this problem?

**Comment by cdcseacave** [comment]
After a lot of trial and error I think the gradient is computing well now, however I still do not know a good procedure to tackle this task: how to modify/add the derivative in the backward step such that it reflects the forward math changes, and how to best test the changes (I tried using `torch.autograd.gradcheck` but first of all I had to convert everything to doubles otherwise it does lack accuracy, and even like that it is not obvious if there is a problem how do I spot where in the code it occurs).

**Comment by amaysaxena** [comment]
Hi @cdcseacave, were you able to get `torch.autograd.gradcheck` to pass after converting everything to double? I'm trying to do something similar but having trouble even getting gradcheck to pass on the existing functionality, particularly with `rasterization_2dgs`.

**Comment by cdcseacave** [comment]
Yes, it was a bit tricky but it worked. However I only tried on the main gsplat pipeline, not on 2dgs variant
